src/ows/src/main/java/org/geoserver/ows/Dispatcher.java

```diff
@@ -1701,17 +1701,34 @@ void exception(Throwable t, Service service, Request request) {
                 logger.log(Level.FINER, "", t);
             }
         } else {
-            logger.log(Level.SEVERE, "", t);
+            String errorCode = null;
+            String message = t.getMessage();
+            Level level = Level.SEVERE;
 
             // unwind the exception stack until we find one we know about
             Throwable cause = t;
             while (cause != null) {
                 if (cause instanceof ServiceException) {
+                    errorCode = ((ServiceException) cause).getCode();
                     break;
                 }
-
                 cause = cause.getCause();
             }
+            if (errorCode == null) {
+                // Some pieces of code catch the exception and add it to the error list
+                // (e.g. KvpUtils during parsing) without rethrowing it or wrapping it
+                // since the service might not be known in advance. Therefore, there
+                // might be the case that it isn't a ServiceException and therefore no
+                // errorCode available. Let's use the error message in that case.
+                errorCode = message;
+            }
+
+            if (errorCode != null && errorCode.toUpperCase().startsWith("INVALID")) {
+                // Log Exceptions dealing with "Invalid" to INFO
+                level = Level.INFO;
+            }
```

Comment:
errorCode is currently used only to determine the logging level, but one might get fooled by the variable later and we might end up having an exception message in the OWS response... how about using directly the message in the logic that determines the error message instead? Something like:

```java
if ((errorCode != null && errorCode.toUpperCase().startsWith("INVALID")) || (errorCode == null && message != null && message.toUpperCase().startsWith("INVALID")))
```

Reply:
@aaime 
Sounds good. fixed